src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/IOperationClassWriter.cs

```diff
@@ -391,7 +399,230 @@ private void WriteClasses()
 
         private void WriteClassNew(AbstractNode type)
         {
-            // PROTOTYPE(iop): implement
+            WriteLine("#nullable enable");
+            var allProps = GetAllProperties(type);
+            bool hasSkippedProperties = !GetAllProperties(type, includeSkipGenerationProperties: true).SequenceEqual(allProps);
+            var ioperationProperties = allProps.Where(p => IsIOperationType(p.Type)).ToList();
+            var publicIOperationProps = ioperationProperties.Where(p => !p.IsInternal).ToList();
+            string typeName = type.Name[1..];
+            var lazyChildren = @"_lazyChildren";
+            var hasType = false;
+            var hasConstantValue = false;
+            var multipleValidKinds = (type.OperationKind?.Entries?.Where(e => e.EditorBrowsable != false).Count() ?? 0) > 1;
+
+            IEnumerable<Property>? baseProperties = null;
+            if (_typeMap[type.Base] is { } baseNode)
+            {
+                baseProperties = GetAllProperties(baseNode);
+            }
+
+            var @class = type.IsAbstract ? $"Base{typeName}" : typeName;
+            var @base = type.Base[1..];
+            if (@base != "Operation")
+            {
+                @base = $"Base{@base}";
+            }
+
+            writeClassHeader(type.IsAbstract ? "abstract" : "sealed", @class, @base, type.Name);
+
+            if (type is Node node)
+            {
+                if (publicIOperationProps.Count != 0)
+                {
+                    WriteLine($"private IEnumerable<IOperation> {lazyChildren};");
+                }
+
+                hasType = node.HasType;
+                hasConstantValue = node.HasConstantValue;
+            }
+            else
+            {
+                node = null!;
```

Comment:
I understand this is due to patterns rules, but that's painful :-/

Reply:
I did not realize this works, and I don't know how I feel about it.